documentation/src/main/asciidoc/userguide/appendices/LegacyBasicTypeResolution.adoc

```diff
@@ -0,0 +1,346 @@
+
+.Java 8 BasicTypes
+[cols="<.^,<.^,<.^,<.^",options="header",]
+|=================================================================================================
+|Hibernate type (org.hibernate.type package) |JDBC type |Java type |BasicTypeRegistry key(s)
+|DurationType |BIGINT |java.time.Duration |Duration, java.time.Duration
+|InstantType |TIMESTAMP |java.time.Instant |Instant, java.time.Instant
+|LocalDateTimeType |TIMESTAMP |java.time.LocalDateTime |LocalDateTime, java.time.LocalDateTime
+|LocalDateType |DATE |java.time.LocalDate |LocalDate, java.time.LocalDate
+|LocalTimeType |TIME |java.time.LocalTime |LocalTime, java.time.LocalTime
+|OffsetDateTimeType |TIMESTAMP |java.time.OffsetDateTime |OffsetDateTime, java.time.OffsetDateTime
+|OffsetTimeType |TIME |java.time.OffsetTime |OffsetTime, java.time.OffsetTime
+|ZonedDateTimeType |TIMESTAMP |java.time.ZonedDateTime |ZonedDateTime, java.time.ZonedDateTime
+|=================================================================================================
+
+.Hibernate Spatial BasicTypes
+[cols="<.^,<.^,<.^,<.^",options="header",]
+|=================================================================================================
+|Hibernate type (org.hibernate.spatial package) |JDBC type |Java type |BasicTypeRegistry key(s)
+|JTSGeometryType |depends on the dialect | com.vividsolutions.jts.geom.Geometry |jts_geometry, and the class names of Geometry and its subclasses
+|GeolatteGeometryType |depends on the dialect | org.geolatte.geom.Geometry |geolatte_geometry, and the class names of Geometry and its subclasses
+|=================================================================================================
+
+[NOTE]
+====
+To use the Hibernate Spatial types, you must add the `hibernate-spatial` dependency to your classpath _and_ use an `org.hibernate.spatial.SpatialDialect` implementation.
+
+See the <<chapters/query/spatial/Spatial.adoc#spatial,Spatial>> chapter for more details.
+====
+
+
+
+[[basic-registry]]
+=== BasicTypeRegistry
+
+We said before that a Hibernate type is not a Java type, nor an SQL type, but that it understands both and performs the marshalling between them.
+But looking at the basic type mappings from the previous examples,
+how did Hibernate know to use its `org.hibernate.type.StringType` for mapping for `java.lang.String` attributes,
+or its `org.hibernate.type.IntegerType` for mapping `java.lang.Integer` attributes?
+
+The answer lies in a service inside Hibernate called the `org.hibernate.type.BasicTypeRegistry`, which essentially maintains a map of `org.hibernate.type.BasicType` (an `org.hibernate.type.Type` specialization) instances keyed by a name.
+
+We will see later, in the <<basic-type-annotation>> section, that we can explicitly tell Hibernate which BasicType to use for a particular attribute.
```

Comment:
See this is something I am actually on the fence about.  All of a sudden things get hard to read because every fifth word is rendered differently.  

Like with Javadoc, I've seen the recommendation to not `{@link}` all references to a given class; like if you reference it 50 times in the same Javadoc block, just link once.

I think there is some validity in that.  Even more so I'd say in asciidoc because the different rendering sometimes really gets in the way of understanding what the words are actually trying to say.

It's a balance for sure